dartagnan/src/main/java/com/dat3m/dartagnan/program/event/lang/llvm/LlvmStore.java

```diff
@@ -0,0 +1,68 @@
+package com.dat3m.dartagnan.program.event.lang.llvm;
+
+import com.dat3m.dartagnan.expression.ExprInterface;
+import com.dat3m.dartagnan.expression.IExpr;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.core.MemEvent;
+import com.dat3m.dartagnan.program.event.core.utils.RegReaderData;
+import com.dat3m.dartagnan.program.event.visitors.EventVisitor;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableSet;
+
+import static com.dat3m.dartagnan.program.event.Tag.C11.*;
+import static com.dat3m.dartagnan.program.event.Tag.*;
+
+public class LlvmStore extends MemEvent implements RegReaderData {
+
+    private ExprInterface value;
+
+    public LlvmStore(IExpr address, ExprInterface value, String mo){
+        super(address, mo);
+        Preconditions.checkArgument(!mo.isEmpty(), "LLVM events cannot have empty memory order");
+        Preconditions.checkArgument(!mo.equals(MO_ACQUIRE) && !mo.equals(MO_ACQUIRE_RELEASE),
+        		getClass().getName() + " cannot have memory order: " + mo);
+        this.value = value;
+        addFilters(ANY, VISIBLE, MEMORY, WRITE, REG_READER);
+    }
+
+    private LlvmStore(LlvmStore other){
+        super(other);
+        this.value = other.value;
+    }
+
+    @Override
+    public ImmutableSet<Register> getDataRegs(){
+        return value.getRegs();
+    }
+
+    @Override
+    public String toString() {
+        return "llvm_store(*" + address + ", " +  value + (mo != null ? ", " + mo : "") + ")\t### LLVM";
+    }
```

Comment:
`mo` cannot be `null` anymore, can it? Together with the fact that `mo` has to be also non-empty (due to this being an atomic), the checks there are useless (probably the same for all other atomic events).